Datadog.Serverless.Compat/DatadogServerlessCompat.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.Logging;
+
+namespace Datadog.Serverless.Compat
+{
+    public enum CloudEnvironment
```

Comment:
Avoid making this `public` unless we need to. Once it's part of the public API we'll want to avoid making breaking changes.
```suggestion
    internal enum CloudEnvironment
```